### ISSUE TITLE
About the estimating of blur kernel

### ISSUE THREAD

**Issue by Sheldon04** [csdwren/SelfDeblur#15]
I wanna know the reason why you use fcn to estimate the blur kernel instead of directly estimate the kernel by CNN from a blur image? Thank you professor.

**Comment by csdwren** [comment]
CNN is designed for generating natural images. But the distribution of blur kernels are quite different from natural images. Thus CNN is not a good choice for estimating blur kernel. You can replace fcn as CNN, and find that CNN can estimate coarse shape of blur kernel but it is quite inferior to fcn.

**Comment by Sheldon04** [comment]
> CNN is designed for generating natural images. But the distribution of blur kernels are quite different from natural images. Thus CNN is not a good choice for estimating blur kernel. You can replace fcn as CNN, and find that CNN can estimate coarse shape of blur kernel but it is quite inferior to fcn.

yeah, I’ve done some work on that and found that the blur core is always surrounded by few magazines

**Comment by Sheldon04** [comment]
how do you know the size of blur kernel?

**Comment by csdwren** [comment]
> how do you know the size of blur kernel?

Generally, it is assumed as a known value. If not, you can set a relatively large value to cover possible blur sizes.

**Comment by Sheldon04** [comment]
Thank you!!! And one more thing, when i was running your model, i found that the result of manmade_05_kernel_03 in lai datasets is low in ssim and psnr(using the method you've supplied), but actually its visual result is quiet good. Then I compared the manmade_05_kernel_03 in datasets/lai and found there is a marked dislocation for this blur img. But in your results, it seems has no dislocation, why is it?

**Comment by csdwren** [comment]
The SelfDeblur actually has some randomness, such as input noise, parameter initialization, noise  perturbation in each iteration etc. I only fixed the input noise, but I cannot guarantee the same results with the paper. Even you run xxx_reprocude.py to load my trained parameters, the results may be slightly different from the paper, since the random noise perturbation.
